src/main/java/io/pravega/connectors/flink/dynamic/table/FlinkPravegaDynamicDeserializationSchema.java

```diff
@@ -47,30 +50,92 @@ public void open(InitializationContext context) throws Exception {
     }
 
     @Override
-    public RowData extractEvent(EventRead<RowData> eventRead) {
-        RowData rowData = eventRead.getEvent();
-        if (metadataKeys.size() == 0 || rowData == null) {
-            return rowData;
-        }
+    public RowData deserialize(byte[] message) throws IOException {
+        return this.nestedSchema.deserialize(message);
+    }
+
+    @Override
+    public void deserialize(byte[] message, Collector<RowData> out) throws IOException {
+        this.nestedSchema.deserialize(message, out);
+    }
+
+    @Override
+    public RowData deserialize(byte[] message, EventRead<ByteBuffer> eventRead) throws IOException {
+        throw new IllegalStateException("To add metadata, use OutputCollector");
```

Comment:
The error message is stil not clear, we can just say `Please invoke FlinkPravegaDynamicDeserializationSchema#deserialize(byte[], EventRead<ByteBuffer>, Collector<RowData>) instead.`